examples/relative_hydration.py

```diff
@@ -0,0 +1,197 @@
+        'lambda_electrostatics' : '2*(lambda - 0.5) * step(lambda - 0.5)',
+        'lambda_bonds' : 'lambda',
+        'lambda_angles' : 'lambda',
+        'lambda_torsions' : 'lambda'
+    }
+
+    w_f = np.zeros(n_iterations)
+    w_r = np.zeros(n_iterations)
+
+    #make the alchemical integrators:
+    forward_integrator = NCMCGHMCAlchemicalIntegrator(temperature, hybrid_system, functions, nsteps=ncmc_nsteps, direction='insert')
+    forward_context = openmm.Context(hybrid_system, forward_integrator, platform)
+    print("Minimizing for forward protocol...")
+    forward_context.setPositions(initial_hybrid_positions)
+    for parm in functions.keys():
+        forward_context.setParameter(parm, 0.0)
+
+    openmm.LocalEnergyMinimizer.minimize(forward_context, maxIterations=10)
+
+    initial_state = forward_context.getState(getPositions=True, getEnergy=True)
+    print("The initial energy after minimization is %s" % str(initial_state.getPotentialEnergy()))
+    initial_forward_positions = initial_state.getPositions(asNumpy=True)
+    equil_positions = simulate_hybrid(hybrid_system,functions, 0.0, initial_forward_positions)
+
+    print("Beginning forward protocols")
+    #first, do forward protocol (lambda=0 -> 1)
+    with progressbar.ProgressBar(max_value=n_iterations) as bar:
+        for i in range(n_iterations):
+            equil_positions = simulate_hybrid(hybrid_system, functions, 0.0, equil_positions)
+            forward_context.setPositions(equil_positions)
+            forward_integrator.step(ncmc_nsteps)
+            w_f[i] = -1.0 * forward_integrator.getLogAcceptanceProbability(forward_context)
+            bar.update(i)
+
+    del forward_context, forward_integrator
+
+    reverse_integrator = NCMCGHMCAlchemicalIntegrator(temperature, hybrid_system, functions, nsteps=ncmc_nsteps, direction='delete')
+
+    print("Minimizing for reverse protocol...")
+    reverse_context = openmm.Context(hybrid_system, reverse_integrator, platform)
+    reverse_context.setPositions(initial_hybrid_positions)
+    for parm in functions.keys():
+        reverse_context.setParameter(parm, 1.0)
+    openmm.LocalEnergyMinimizer.minimize(reverse_context, maxIterations=10)
+    initial_state = reverse_context.getState(getPositions=True, getEnergy=True)
+    print("The initial energy after minimization is %s" % str(initial_state.getPotentialEnergy()))
+    initial_reverse_positions = initial_state.getPositions(asNumpy=True)
+    equil_positions = simulate_hybrid(hybrid_system,functions, 1.0, initial_reverse_positions, nsteps=1000)
+
+    #now, reverse protocol
+    print("Beginning reverse protocols...")
+    with progressbar.ProgressBar(max_value=n_iterations) as bar:
+        for i in range(n_iterations):
+            equil_positions = simulate_hybrid(hybrid_system,functions, 1.0, equil_positions)
+            reverse_context.setPositions(equil_positions)
+            reverse_integrator.step(ncmc_nsteps)
```

Comment:
I think have to reset the `reverse_integrator` each iteration with a call to [`reverse_integrator.reset()`](https://github.com/choderalab/perses/blob/master/perses/annihilation/ncmc_switching.py#L954-L967).